### DISCUSSION TITLE
Adding buttons to note template instead of toolbar?

### DISCUSSION THREAD

**Discussion by ecating** [TriliumNext/Trilium#2049] [Q&A] [answered]
First off, I'm really enjoying Trilium and appreciate the Wiki! 

I know that I can add buttons to the toolbar (very neat, btw), but am wondering if there is anything in the API that would let me add a button to a note itself - specifically to a note template so it shows up in each of that note type. 

Thanks much!

Emma

**Comment by zadam** [comment, marked as answer]
Hi, check [custom widget](https://github.com/zadam/trilium/wiki/Custom-widget). In that example's `refreshWithNote()` it checks for a specific note type and based on that renders some dynamic content. Same pattern could be used to show e.g. buttons or whatever else ...

**Reply by ecating** [reply]
Thanks!
